### ISSUE TITLE
All problems seem to be missing basic instructions 

### ISSUE THREAD

**Issue by kaushalmodi** [exercism/nim#79]
At present, I need to read the test suite to figure out the complete problem statement.

Each problem README should contain information like:

- Name of the to-be-submitted .nim file. For example, the "Word Count" problem, needs the file to be name `wordcount.nim`, not `WordCount.nim` or anything else, because the test suite is importing using `import wordcount`.
- Name of procs and anything else to be made exportable from the submitted library. For example, in the "Leap" problem, the expected exported proc name is `isLeapYear` while the libary name expected is `leap`. There is not way for a user to guess this.. they **have** to read and understand the test suite.
- In addition to the naming of procs, also mention in the README the required signature for those procs.
- Another example.. in the "Bob" problem, the specs are loose like "He answers 'Whoa, chill out!' if you yell at him.".. what is yelling? A better programming spec would be "check if all chars are upper case".

Basically, the submission ready code should be able to be created by reading just the README. At the moment, the README's provide almost no information, and I need to reverse engineer the tests.

**Comment by amscotti** [comment]
@kaushalmodi thanks for your feedback, I think for some of the issues we can add a "Hint" file to help people understand what the filename should be for the import to work properly. Part of this is understanding Nim's naming conventions but for someone new to Nim we should give a helping hand with this. This is a good idea, thanks for pointing this out!

As far as the details of the readme, this comes from the `problem-specifications` at https://github.com/exercism/problem-specifications. The Nim track uses this repository to create the readme files for the exercises, this helps to ensure that all readme is the same from track to track. I think taking time to understand the test suite to get a better understanding of what is needed for your code is part of the learning experience and also follows practices of test-driven development.

As the problem-specifications is another open project under Exercism, I would encourage you to share your feedback with them also.

Thanks again for your feedback! I'm going to look into adding hint files or some other way to help people know what the filename should be.

**Comment by amscotti** [comment]
I think all issues have been addressed here, or other issues created. Issue #87 will fix the issue with the proc and file naming as this will give a file for someone to work in without the need to create it themselves.

The Bob exercise was updated with PR #80 and word-count with PR #81.

Again thanks for your feedback on the Nim track! I'm going to close this issue.
